packages/react-utilities/src/compose/resolveShorthand.ts

```diff
@@ -1,30 +1,37 @@
 import { isValidElement } from 'react';
 import { ObjectShorthandProps, ShorthandProps } from './types';
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface ResolveShorthandOptions<Props extends Record<string, any>, Optional extends boolean = true> {
```

Comment:
Why is there a generic for Optional?

Reply:
It's impossible to infer optional value without a generic, and inference is required to decided if the return value will be nullable or not, otherwise all signatures would have to carry `undefined`